lib/useOnyx.ts

```diff
@@ -321,11 +323,19 @@ function useOnyx<TKey extends OnyxKey, TReturnValue = OnyxValue<TKey>>(
         [key, options?.initWithStoredValues, options?.reuseConnection, checkEvictableKey],
     );
 
+    const getSnapshotDecorated = useMemo(() => {
+        if (!GlobalSettings.isPerformanceMetricsEnabled()) {
+            return getSnapshot;
+        }
+
+        return decorateWithMetrics(getSnapshot, 'useOnyx.getSnapshot');
```

Comment:
I think this implementation will fail once `getSnapshot` get recreated because we are throwing an Error [here](https://github.com/Expensify/react-native-onyx/blob/main/lib/metrics.ts#L24-L26) if we decorate a method with the same alias. Probable solutions would be creating a method to remove the alias from `decoratedAliases` set, or changing `decorateWithMetrics` implementation to not throw errors in this case.

Reply:
https://github.com/Expensify/react-native-onyx/blob/c25f9a07cb62b04c9e6c9ae8f6a564230d3c5a85/lib/metrics.ts#L25

Yeah I think we can safely just early return the function that was passed in without trying to re-decorate it.

Reply:
Went the other way - removed the memory set completely. I believe it's not needed, we only decorate things once for onyx utilities & allow overriding for compatibility with React.